packages/components/src/radio-control/index.tsx

```diff
@@ -88,7 +88,13 @@ export function RadioControl(
 							onChange={ onChangeValue }
 							checked={ option.value === selected }
 							aria-describedby={
-								!! help ? `${ id }__help` : undefined
+								// TODO: will improve if we like this solution
+								// eslint-disable-next-line no-nested-ternary
+								!! option.helpText
+									? `${ id }-${ index }-help`
+									: !! help
+									? `${ id }__help`
+									: undefined
```

Comment:
Currently, the option's help text is used to describe the associated `input` element — although I'm open to alternatives, if you have any

Reply:
> Since the help text is not part of the `label`, clicking it won't select the associated radio.

I'd say this is the desired behavior. Clicking on `help` text doesn't (and I'd say shouldn't) focus the control (same in InputControl, CheckboxControl, etc).

Reply:
This could be a good opportunity to fix #52890, by changing this component to a proper `fieldset` markup and associating the group `help` to that.

If that's out of scope for this PR though, it might be better to keep both the group-level and option-level `help`s associated if they both exist (apparently you can specify [multiple space-delimited ids](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#values)).

Reply:
> This could be a good opportunity to fix https://github.com/WordPress/gutenberg/issues/52890, by changing this component to a proper fieldset markup and associating the group help to that.

That would be great, but do you know if `BaseControl` supports using `fieldset` ?

> If that's out of scope for this PR though, it might be better to keep both the group-level and option-level helps associated if they both exist (apparently you can specify [multiple space-delimited ids](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#values)).

That's what I've done for now.

Reply:
> That would be great, but do you know if `BaseControl` supports using `fieldset` ?

No, we currently need to do custom implementations using `BaseControl.VisualLabel`. But baking `fieldset` support into BaseControl might be something to consider! (Not immediately sure if that's a good idea)

Reply:
> No, we currently need to do custom implementations using `BaseControl.VisualLabel`.

Ok, this definitely sounds to me like a task that should have its dedicated PR. I will not work on this in this PR.

> But baking `fieldset` support into BaseControl might be something to consider! (Not immediately sure if that's a good idea)

https://github.com/WordPress/gutenberg/pull/63751#discussion_r1688603767